[stable/kong] major breaking change: dbless and KIC by-default

hbagdi opened the pull request:
Also, a changelog section has been started that should be updated for
each and every release.

Signed-off-by: Harry Bagdi <harrybagdi@gmail.com>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->
#### Special notes for your reviewer:

We are starting a new Changelog section with this release to easily communicate changes with each release to end-users. The changelog will be available on the web at the following location:
https://hub.helm.sh/charts/stable/kong

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

shashiranjan84 commented on stable/kong/README.md:
```diff
@@ -139,7 +139,7 @@ for the service definition to work properly.
 Kong has a choice of either Postgres or Cassandra as a backend datatstore.
 This chart allows you to choose either of them with the `env.database`
-parameter. Postgres is chosen by default.
+parameter. The default is to use Kong without a database.
```
This whole paragraph needs to be updated to make sense.

hbagdi committed:
[stable/kong] major breaking change: dbless and KIC by-default

Also, a changelog section has been started that should be updated for
each and every release.

Signed-off-by: Harry Bagdi <harrybagdi@gmail.com>

shashiranjan84 reviewed (approved):
/lgtm
